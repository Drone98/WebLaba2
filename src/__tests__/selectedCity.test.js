import React from 'react'
import renderer from 'react-test-renderer'
import SelectedCity from '../classes/selectedCity'
import '@babel/polyfill'
import {applyMiddleware, createStore} from "redux";
import reducer from "../storage/reducer"
import thunk from "redux-thunk";
import {Provider} from "react-redux";

const {act} = renderer;

const data = (url, params) => {
    return Promise.resolve({
        "data": {
            "coord": {
                "lon": params.params.lon,
                "lat": params.params.lat
            },
            "weather": [
                {
                    "description": "overcast clouds",
                    "icon": "04d"
                }
            ],
            "main": {
                "temp": 0,
                "pressure": 1000,
                "humidity": 80
            },
            "wind": {
                "speed": 5
            }
        }
        }
    );
};

let store;

beforeEach(() => {
        let persistedState = [{name: "murmansk"}, {name: "moscow"}];
        store = createStore(reducer, persistedState, applyMiddleware(thunk));
    }
);

it('Rendered selected city',async () => {
    let axios = {
        get: data
    };
    let component = await renderer.create(
        <Provider store={store}>
            <SelectedCity axios={axios} name={"murmansk"}/>
        </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot()
});

it('Loading', () => {
    let axios = {
        get: data
    };
    let component = renderer.create(
        <Provider store={store}>
            <SelectedCity axios={axios} name={"murmansk"} loadState={true}/>
        </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot()
});

it('Server error', async () => {
    let axios = {
        get: () => {
            return Promise.reject({
                response: {
                    status: 400
                }
            })
        }
    };
    let component = renderer.create(<div/>);
    await act(() => {
        component = renderer.create(
            <Provider store={store}>
                <SelectedCity axios={axios} name={"murmansk"}/>
            </Provider>
        );
    });
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot()

});

it('Network error', async () => {
    let axios = {
        get: () => {
            return Promise.reject({
                response: null
            })
        }
    };

    let component = renderer.create(<div/>);

    await act(() => {
        component = renderer.create(
            <Provider store={store}>
                <SelectedCity axios={axios} name={"murmansk"}/>
            </Provider>
        );
    });
});