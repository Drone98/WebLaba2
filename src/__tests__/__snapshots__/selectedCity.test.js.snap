// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading 1`] = `
<div
  className="loader"
>
  <b>
    Загрузка
  </b>
  <img
    alt=""
    className="loading-icon"
    src=""
  />
</div>
`;

exports[`Rendered selected city 1`] = `
<div
  className="loader"
>
  <b>
    Загрузка
  </b>
  <img
    alt=""
    className="loading-icon"
    src=""
  />
</div>
`;

exports[`Server error 1`] = `
<div
  className="loader"
>
  <b>
    Загрузка
  </b>
  <img
    alt=""
    className="loading-icon"
    src=""
  />
</div>
`;
